code/CloudFlare.php

```diff
@@ -59,241 +63,102 @@ public function getCFCredentials()
-            )
-        );
+        Deprecation::notice('2.0', 'This method will be removed in favor for CloudFlare_Purge functionality');
 
-        $result = $this->purgeRequest($data);
+        $purger = CloudFlare_Purge::create();
+        $purger
+            ->setSuccessMessage('CloudFlare cache has been purged for: ' . $fileOrUrl)
+            ->pushFile($fileOrUrl)
+            ->purge();
 
-        return $this->responseHandler($result, 'CloudFlare cache has been purged for: ' . $fileOrUrl);
+        return $purger->getResponse();
     }
 
     /**
      * Same as purgeSingle with the obvious functionality to handle many
      *
+     * @deprecated This method will be removed in favor for CloudFlare_Purge functionality
+     *
      * @param array $filesOrUrls
      *
      * @return bool
      */
     public function purgeMany(array $filesOrUrls)
     {
-        // fetch zone ID dynamically
-        $zoneId = $this->fetchZoneID();
-
-        $count = count($filesOrUrls);
-
-        if (!$zoneId) {
-            if ($this->config()->log_errors) {
-                error_log("CloudFlareExt: Attempted to purge cache for {$count} files but unable to find Zone ID");
-            }
-
-            return false;
-        }
-
-        $baseUrl = Director::absoluteBaseURL();
-
-        $purgeUrls = array();
-
-        foreach ($filesOrUrls as $fileOrUrl) {
-            $purgeUrls[] = $baseUrl . $fileOrUrl;
-        }
-
-        $data = array(
-            "files" => $purgeUrls
-        );
+        Deprecation::notice('2.0', 'This method will be removed in favor for CloudFlare_Purge functionality');
 
-        $response = $this->purgeRequest($data);
+        $purger = CloudFlare_Purge::create();
+        $purger
+            ->setSuccessMessage("CloudFlare cache has been purged for: {file_count} files")
+            ->setFiles($filesOrUrls)
+            ->purge();
 
-        return $this->responseHandler(
-            $response,
-            "CloudFlare cache has been purged for: {$count} files as a result of updating this page"
-        );
+        return $purger->getResponse();
 
     }
 
     /**
      * Purges everything that CloudFlare has cached
      *
+     * @deprecated This method will be removed in favor for CloudFlare_Purge functionality
+     *
      * @param null $customAlert
      *
      * @return mixed
      */
     public function purgeAll($customAlert = null)
     {
-        $data = array(
-            "purge_everything" => true
-        );
-
-        $response = $this->purgeRequest($data);
+        Deprecation::notice('2.0', 'This method will be removed in favor for CloudFlare_Purge functionality');
 
-        return $this->responseHandler(
-            $response,
-            $customAlert ?: _t(
-                "CloudFlare.PurgedEverything",
-                "Successfully purged <strong>EVERYTHING</strong> from cache."
-            )
-        );
+        return CloudFlareAdmin::singleton()->purge_all();
```

Comment:
I don't like this. If I didn't want the admin model then this would break my purge all capability